Plugins/NativeCamera/Source/NativeCameraImpl.h

```diff
@@ -17,24 +39,18 @@ namespace Babylon::Plugins
         };
 
         struct ImplData;
-        
+
         Impl(Napi::Env env, bool overrideCameraTexture);
         ~Impl();
-        arcana::task<CameraDimensions, std::exception_ptr> Open(uint32_t maxWidth, uint32_t maxHeight, bool frontCamera);
+        arcana::task<CameraDimensions, std::exception_ptr> Open(std::shared_ptr<CameraDevice> cameraDevice, std::shared_ptr<CameraTrack> track);
+
+        std::vector<std::shared_ptr<CameraDevice>> GetCameraDevices();
```

Comment:
Following along the same vein as the other comments should ultimately be std::vector<CameraDevice>&